Amplify/Categories/DataStore/Model/Internal/Schema/AuthRule.swift

```diff
@@ -22,6 +22,12 @@ public enum ModelOperation {
     case update
     case delete
     case read
+    // granular read access
+    case `get`
+    case list 
+    case sync 
+    case listen 
+    case search
```

Comment:
I bet a lot of logic is determined by checking if the operation is `.read` and does something. Now that we allow compiling the more granular "read" types, that means all logic based on "read" should now be updated to

1. If the previous check for read expected all granular read access then update the code to "if operation == read || operations.containsAll(list, sync, listen, search)"  
2. If the previous check for read expected just the sync API (for example DataStore's sync operation), then code can be updated to "if operation == read || operation == sync)"
3. If previous check for read expected that the subscription API is used then --> "operation == read or operation ==  "listen"